# Exome-seq
**tools preparation**

softwares available to conda:
    
    conda install sra-tools
    conda install fastqc
    conda install fastx_toolkit
    conda install trim-galore
    conda install bwa
    conda install samtools
    conda install snpeff
    conda install bcftools
    conda install qualimap
    
 softwares not available to conda:
 
    wget -c -t 0 https://github.com/broadinstitute/gatk/releases/download/4.2.2.0/gatk-4.2.2.0.zip
    unzip gatk-4.2.2.0.zip
    cd gatk-4.2.2.0
    ./gatk
    echo 'export PATH="/home/students/y.dong/tool/gatk-4.2.2.0/:$PATH"' >> ~/.bashrc
    source ~/.bashrc
    
**step 1: Download SRA files, high-throughput raw sequencing data.**
  method one:using the prefetch function of the SRA-tools.
    E.g:download SRR3589957-SRR3589965(RNAseq of *Homo sapiens* 293 cell line)
  
    for i in `seq 57 65`;
    do    
    prefetch SRR35899${i}
    done
  method two:using the FTP protocol.
    E.g:download SRR5422017-SRR5422019(RNAseq of *Saccharomyces cerevisiae*)
    
    for i in `seq 7 9`;
    do
    wget -c -t 0 ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR542/SRR542201${i}/SRR542201${i}.sra	#new
    done
  method three:using Entrez Direct package.
    E.g:download all SRA files under the SRP091493 project number and unzip those SRA file.
  
    $ esearch -db sra -query SRP091493 | efetch --format runinfo | cut -d ',' -f 1 | grep SRR | xargs fastq-dump --split-files â€“bzip2
   method four:using aspera.
    E.g:all SRA files in file a can be downloaded. File a can be directly accessed by the NCBI project into the accession list option in all runs.
    
    $ prefetch -t ascp -a "/home/user/.aspera/connect/bin/ascp|/home/user/.aspera/connect/etc/asperaweb_id_dsa.openssh" --option-file SRR_Acc_List.txt -O ./
**step 2:Unzip the SRA file and do quality control**

    $ bash SRA-QC.sh

**step 3: Remove adapter and filter low-quality reads**

    $ bash clipper.sh

**step 4:Download the genome file and annotation file for the corresponding specie**
  
  method one: find the corresponding genome in NCBI, copy the download link, and download it directly using the wget command.
  E.g:download genome of *Drosophila_melanogaster*,and the annotation file is gff,the same download method.
  
    $ wget -c -t 0 ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/215/GCF_000001215.4_Release_6_plus_ISO1_MT/GCF_000001215.4_Release_6_plus_ISO1_MT_genomic.fna.gz
    
  method two: download genome, annotation, and indexes from iGenomes(https://support.illumina.com/sequencing/sequencing_software/igenome.html).
  
**step 5:Use the mapping tool to build index of the corresponding species genome**
  E.g:Use BWA to build an index file for the human genome. If we download genome files from iGenomes, this step can be ommited.
  
    $ bwa index ./Homo_sapiens.fasta -p genome
    
**step 6:Map to genome.**
There are many different sequence alignment tools, such as STAR, hisat2, BWA, Tophat, bowtie2...but BWA is widely used in exome-seq mapping.
  
    $ bash bwa.sh
    
**step 7:Use samtools to convert the sam files generated by the alignment to bam files and sort them. Then we can use samtools to statistic the mapping results**

    $ bash samtools.sh
    
  Let's check the BAM file:
    
    $ samtools mpileup SRRXXXXXXXX.bam |head -100|tail -3
    
**step 8: Find variants.**
  Firstly, we need download the annotation files of variants from GATK. I use *lftp* to download. If we aren't root, we can install *lftp* by conda. It's best to install tools in another virtual environment built by conda, not in the base environment, so as not to cause conda damage. Because the libc.so.6 version of the workstation is often low.
  
    $ lftp ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/
    > mirror hg38
    > exit
    $ tree -h
    $ gunzip *.gz
  
  The easiest way to find variants (not precise):
  
    $ samtools mpileup -ugf /DYY/gatk/hg38/Homo_sapiens_assembly38.fasta *.bam | bcftools call -vmO z -o var_out.vcf.gz
    
  The complete way using GATK. Firstly, we need to mark PCR duplicate sequences. If the library is paired-end, we need fix the mate information of the pair-end. And if we forget add read groups in BAM file by BWA, we can use *AddOrReplaceReadGroups* to add read group information, run '$bash add_read_group.sh'. Then base quality score recalibration is necessary. 

    $ bash gatk_dup_bqsr.sh
  
  If 
    
    





